qiskit/transpiler/passes/layout/csp_layout.py

```diff
@@ -109,6 +136,10 @@ def run(self, dag):
                      qubits.index(gate.qargs[1])))
         edges = set(self.coupling_map.get_edges())
 
+        if not self.strict_direction:
+            cxs = {frozenset(cx) for cx in cxs}
```

Comment:
This feel unnecessary as you end up iterating over the set of two qubit ops twice. Why not just have the if block around the loop and do this iteratively as you build cx?